cli/schema/cypress.schema.json

```diff
@@ -276,6 +276,16 @@
       "type": "boolean",
       "default": false,
       "description": "Enables including elements within the shadow DOM when using querying commands (e.g. cy.get(), cy.find()). Can be set globally in cypress.json, per-suite or per-test in the test configuration object, or programmatically with Cypress.config()"
+    },
+    "component": {
+      "type": "object",
```

Comment:
I believe there is a way to be nicer on the schema here without repeating ourselves.

```json
{
  "title": "JSON schema for the https://cypress.io Test Runner's configuration file. Details at https://on.cypress.io/configuration",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "definitions": {
    "cypressConfig": {
      // ...
      "component": {
          "description": "Any component runner specific overrides",
          "$ref": "#/definitions/cypressConfig"
        },
        "e2e": {
          "description": "Any e2e runner specific overrides",
          "$ref": "#/definitions/cypressConfig"
        }
    }
  },
  "$ref": "#/definitions/cypressConfig" 
}

Reply:
I made a commit and tested the results. It seems to be working smoothly.